numerical_methods/matrix_transpose.c

```diff
@@ -0,0 +1,135 @@
+#include <assert.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdlib.h>
```

Comment:
Please add a one-line description of what the library/header is for (**see the example below**).

```c
#include <stdio.h>     /// for IO operations
#include <assert.h>    /// for assert
```